jbehave-core/src/main/java/org/jbehave/core/model/TableTransformers.java

```diff
@@ -204,4 +222,100 @@ public String transform(String tableAsString, TableParsers tableParsers, TablePr
             return tableAsString.replace(replacing, replacement);
         }
     }
+
+    public static class SelfReferencing implements TableTransformer {
+        private final ParameterControls parameterControls;
+        private final Pattern placeholderPattern;
+
+        public SelfReferencing(ParameterControls parameterControls) {
+            this.parameterControls = parameterControls;
+            placeholderPattern = Pattern.compile(
+                    parameterControls.nameDelimiterLeft() + "(.*?)" + parameterControls.nameDelimiterRight());
+        }
+
+        @Override
+        public String transform(String tableAsString, TableParsers tableParsers, TableProperties properties) {
+            String rowSeparator = properties.getRowSeparator();
+            TableRows rows = tableParsers.parseRows(tableAsString, properties);
```

Comment:
it makes sense to check for non-distinct columns

Reply:
> it makes sense to check for non-distinct columns

We have already this check
https://github.com/jbehave/jbehave-core/blob/master/jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTable.java#L294